planner/core/integration_test.go

```diff
@@ -906,3 +906,17 @@ func (s *testIntegrationSuite) TestIssue16290And16292(c *C) {
 		tk.MustQuery("select count(distinct b) from (select * from t ta union all select * from t tb) t;").Check(testkit.Rows("1"))
 	}
 }
+
+func (s *testIntegrationSerialSuite) TestIssue16837(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int,b int,c int,d int,e int,unique key idx_ab(a,b),unique key(c),unique key(d))")
+	tk.MustQuery("explain select /*+ use_index_merge(t,c,idx_ab) */ * from t where a = 1 or (e = 1 and c = 1)").Check(testkit.Rows(
```

Comment:
can we optimize it to use index merge?

Reply:
It seems to me that adding a filter on `IndexRangeScan` can solve this prob. @zz-jason

Reply:
exactly

Reply:
Currently, we are using this mechanism in `from (a>1 and b>1) or c=1 ` with `key(a,b), key(c)`

Reply:
I will try to solve this later. (after labor day holiday)